analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SupplyParameterFromQueryNeedRoutableComponent_Noncompliant.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.AspNetCore.Components;
+using System;
+
+public class SupplyParameterFromQueryNeedRoutableComponent_Noncompliant : ComponentBase
+{
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public TimeSpan TimeSpan { get; set; } // Noncompliant {{Component parameters can only receive query parameter values in routable components.}}
+    //              ^^^^^^^^
+
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public string MyString { get; set; } // Noncompliant
```

Comment:
I would also add the case where `Parameter` and `SupplyParameterFromQuery` are in the same attribute list: `[Parameter, SupplyParameterFromQuery]`.